libqtile/widget/prompt.py

```diff
@@ -377,15 +378,21 @@ def __init__(self, name="prompt", **config):
                         # can't detect what's a pickle error and what's not.
                         logger.exception("failed to load prompt history")
                         self.history = {x: deque(maxlen=self.max_history)
-                                        for x in self.completers if x}
+                                        for x in self.completers}
+
+                    # self.history of size does not match.
+                    if len(self.history) != len(self.completers):
+                        self.history = {x: deque(maxlen=self.max_history)
+                                for x in self.completers}
+
                     if self.max_history != \
                        self.history[list(self.history)[0]].maxlen:
                         self.history = {x: deque(self.history[x],
                                                  self.max_history)
-                                        for x in self.completers if x}
```

Comment:
Why do you remove all of the `if x` from the dict comprehensions?

Reply:
This ```if x``` occurred error at https://github.com/qtile/qtile/blob/develop/libqtile/widget/prompt.py#L442.
Is it better to resolve this problem by another changings?